etl/main.py

```diff
@@ -1,214 +1,242 @@
+
+            for g_price in gas_prices:
+                places_and_prices[pid][f'{g_price["@type"]}_price'] = float(
+                    g_price['#text'])
+
+    return pd.DataFrame.from_dict(places_and_prices, orient='index')
+
+
+def reverse_geocode(stations_df, geo_gdf):
+    """
+    Performs reverse geocoding on stations_gdf against the DIVA-GIS data to obtain
+    city and state information
+    
+    Returns the stations GeoDataFrame with new columns for city and state
+    """
+    stations_gdf = gpd.GeoDataFrame(stations_df, geometry=gpd.points_from_xy(stations_df.longitude, stations_df.latitude)).set_crs(epsg=4326)
+    stations_geo_gdf = gpd.sjoin(stations_gdf, geo_gdf[['NAME_1', 'NAME_2', 'geometry']])
+    stations_geo_gdf = stations_geo_gdf.rename(columns={'NAME_1': 'state', 'NAME_2': 'city'})
+    return stations_geo_gdf
+
+
+def transform(stations_df, geo_gdf):
+    """
+    This function cleans the gas stations dataframe in order to obtain records
+    with at least one gas type price and correct values 
+
+    It returns the representation of the cleaned dataframe as a list of dictionaries
+    """
+    print('Cleaning and packing data...')
+
+    stations_complete_data_df = stations_df[stations_df['latitude'].notna() & stations_df['longitude'].notna() &
+        (stations_df['regular_price'].notna() | stations_df['premium_price'].notna() | stations_df['diesel_price'].notna())].copy()
+
+    bad_records = stations_complete_data_df[(stations_complete_data_df['regular_price'] <= 1) | 
+        (stations_complete_data_df['diesel_price'] <= 1) | (stations_complete_data_df['premium_price'] <= 1) | 
+        (stations_complete_data_df['regular_price'] >= 40) | (stations_complete_data_df['diesel_price'] >= 40) |
+        (stations_complete_data_df['premium_price'] >= 40)]
+
+    stations_complete_data_df.drop(bad_records.index, inplace=True)
+
+    stations_complete_data_df.index.names = ['id']
+
+    stations_geo_gdf = reverse_geocode(stations_complete_data_df, geo_gdf)
+
+    print(stations_geo_gdf[:10])
+
+    return stations_complete_data_df.reset_index(0).to_dict('records')
+
+
+def load(stations_dict):
+    """
+    Loads the information in the data structure stations_dict to the database
+    """
+    print(f'Connecting to the database...')
+
+    engine = db.create_engine(DB_STRING)
+    metadata = db.MetaData()
+    prices_table = db.Table(
+        'gasoline_price', metadata, autoload=True, autoload_with=engine)
+    stations_table = db.Table(
+        'gasoline_station', metadata, autoload=True, autoload_with=engine)
+
+    with engine.connect() as connection:
+        print('Inserting data...')
+
+        for record in stations_dict:
+            stations_data = {
+                'id': record['id'],
+                'name': record['name'],
+                'longitude': record['longitude'],
+                'latitude': record['latitude'],
+                'town': record['city'],

```

Comment:
There is a  key error when the script run. This is the complete traceback:

![image](https://user-images.githubusercontent.com/44986369/97523617-be9e2980-1970-11eb-871a-e875e943c7bc.png)

![image](https://user-images.githubusercontent.com/44986369/97523597-b47c2b00-1970-11eb-8175-954e43b6010f.png)

Reply:
Corrected.